lib/Frontend/CompilerInvocation.cpp

```diff
@@ -306,11 +304,14 @@ static bool ParseCodeGenArgs(CodeGenOptions &Opts, ArgList &Args, InputKind IK,
   bool Success = true;
 
   unsigned OptLevel = getOptimizationLevel(Args, IK, Diags);
-  if (OptLevel > 3) {
-    Diags.Report(diag::err_drv_invalid_value)
-      << Args.getLastArg(OPT_O)->getAsString(Args) << OptLevel;
-    OptLevel = 3;
-    Success = false;
+  unsigned MaxOptLevel = 3;
+  if (IK == IK_OpenCL){
```

Comment:
I guess there may be people who want to use this branch for not just spir target, but I rest my case.